### ISSUE TITLE
[BUG] - Unit Tests are not cleaning up after itself properly

### ISSUE THREAD

**Issue by happytreees** [vultr/ansible-collection-vultr#59]
**Describe the bug**
After reviewing the account that the tests are run in I noticed a large amount of instances that have not been removed. The behavior of the scripts needs to be updated to ensure that all resources are removed at the end of the run. 

```
+ Cleaning instances
- Deleting ansible-test-58206534-fv-az457-575_vm_for_attachment - ad0a1ee7-d1ba-45a4-a749-55b3d60c6216  
- Deleting ansible-test-29567499-fv-az241-340_vm_for_attachment - 12ed1fda-ffcc-4a8c-9b5f-8aef1a7a77b3  
- Deleting ansible-test-95539053-fv-az264-892_vm_for_attachment - 95e32594-601b-4196-a669-ae54394f806e  
- Deleting ansible-test-30742365-fv-az571-443_vm_for_attachment - 26284b71-f9d2-48f0-b650-608219465dad  
- Deleting ansible-test-41051901-fv-az399-96_vm_for_attachment - d2cbb0f5-d0ee-416c-a202-014d3a955d91  
- Deleting ansible-test-55556826-fv-az259-289_vm_for_attachment - 93ffed8c-cfb3-4000-bae3-6bf2f364388b  
- Deleting ansible-test-53849020-fv-az41-963_vm_for_attachment - 5f37ef4e-fb96-4206-b269-1edcfe9364a7  
- Deleting ansible-test-57154596-fv-az41-963_vm_for_attachment - 330c427b-b69c-4862-bb28-364fdc2c7215  
- Deleting ansible-test-68561648-fv-az264-892_vm_for_attachment - ff51739b-541f-493c-add2-aa4a5f45c7cb  
- Deleting ansible-test-98480117-fv-az554-228_vm_for_attachment - faa416e3-bfcd-4683-acfa-889ee1e4a5ef  
- Deleting ansible-test-98273835-fv-az213-499_vm_for_attachment - a7fc6088-f78b-48f7-964e-535c74d264aa  
- Deleting ansible-test-90207754-fv-az460-75_vm_for_attachment - 691b2843-c4d8-428f-ab69-b9eff6445e32  
- Deleting ansible-test-99445897-fv-az180-114_vm_for_attachment - 5cc8b9cd-4ba1-4628-a8d2-781e7ab1e3ed  
- Deleting ansible-test-14927640-fv-az341-509_vm_for_attachment - cc1c94f5-9eb8-48e4-bd98-83bf40a79296  
- Deleting ansible-test-71858876-fv-az222-453_vm_for_attachment - ce812830-14ea-4b28-b6bd-fe794347c899  
- Deleting ansible-test-72022827-fv-az203-353_vm_for_attachment - 8f02071b-7746-4ec2-9e74-b58bc0a06457  
- Deleting ansible-test-63756813-fv-az208-571_vm_for_attachment - 66e0338c-6a7a-4644-b10f-21838049fdc7  
- Deleting ansible-test-42203396-fv-az83-370_vm_for_attachment - c8fe924e-fb62-4b09-8c50-4970de1ea951  
- Deleting ansible-test-36550713-fv-az345-706_vm_for_attachment - b5026d38-d1ae-4724-9f22-ce7341861f45  
- Deleting ansible-test-32412766-fv-az77-505_vm_for_attachment - 35c3fd4e-ad41-4619-a0e3-1f969e7650f6  
- Deleting ansible-test-47962799-fv-az203-353_vm_for_attachment - cb7812ab-2752-4762-9cba-4c848bfe833c  
- Deleting ansible-test-89186448-fv-az377-906_vm_for_attachment - 41f88eec-422c-4602-8b0a-4a2042877bc9  
- Deleting ansible-test-15512236-fv-az77-505_vm_for_attachment - ca7d6368-032e-457b-8a40-cbb84defa219  
- Deleting ansible-test-44393646-fv-az154-459_os1 - ec76e7b2-94cf-4de0-aa2d-63e410601589  
- Deleting ansible-test-59959984-fv-az334-647_os1 - 7d2519f6-e870-4d69-8be0-85a10740cfba  
- Deleting ansible-test-83430314-fv-az91-21_os1 - 77ee4563-4ed9-49c1-b9c4-d2c8c616c8cc  
- Deleting ansible-test-71595837-fv-az292-792_os1 - b4722340-605d-4117-9a71-dd64470514f4  
- Deleting ansible-test-55851971-fv-az391-222_os1 - a97544ad-7e54-48db-bd57-eb6e094e5fef  
- Deleting ansible-test-55851971-fv-az391-222_info1 - f7f8ce9c-d464-4de5-b8e9-30eed9867f27  
- Deleting ansible-test-55851971-fv-az391-222_info2 - f06734e1-e941-4f4a-99b1-c6a32fa54387  
+ Instances cleaned
+ No bare metals to remove
+ No load balancers to remove
+ No snapshots to remove
+ Cleaning block storage
- Deleting ansible-test-fv-az77-681-61434800-volume - d5e4e4b3-7403-40da-8a4d-7176ad46ada3  
- Deleting ansible-test-36038483-fv-az127-139-volume - 96d4b858-6381-4db5-8f13-b285d8548825  
- Deleting ansible-test-40355126-fv-az246-917-volume - 4e60daec-a2cf-46c9-98ef-7b7f38721680  
- Deleting ansible-test-76871637-fv-az194-118-volume - 8c5f28b4-2846-45d0-8109-636aa9d829ac  
- Deleting ansible-test-13579941-fv-az183-630-volume - a008662e-f064-4661-b21c-003d8e58f2ed  
- Deleting ansible-test-98693036-fv-az246-949-volume - 1dc4d6fd-a5b0-4942-8297-274920c461d4  
+ Block Storage cleaned
```

**Expected behavior**
Clean the resources created regardless of whether or not the tests passed/failed.

**Comment by resmo** [comment]
@happytreees thanks for the cleanup, having a look asap.

Many seem related to a test currently disabled https://github.com/vultr/ansible-collection-vultr/blob/main/tests/integration/targets/block_storage/tasks/main.yml#L6 but the main issue is, that the cleanup is currently not working. https://github.com/vultr/ansible-collection-vultr/blob/main/tests/integration/targets/cleanup/defaults/main.yml#L3

**Comment by resmo** [comment]
```
 failed: [testhost] (item={'label': 'ansible-test-55941776-fv-az186-460_info2', 'hostname': 'info2', 'plan': 'vc2-1c-1gb', 'tags': ['three', 'four'], 'region': 'cdg', 'os': 'Debian 11 x64 (bullseye)'}) => {"ansible_loop_var": "instance", "changed": false, "fetch_url_info": {"body": "{\"error\":\"Unable to destroy server: Unable to remove VM: Server is currently locked\",\"status\":500}", "connection": "close", "content-type": "application/json", "date": "Tue, 28 Mar 2023 21:34:15 GMT", "msg": "HTTP Error 500: Internal Server Error", "server": "nginx", "status": 500, "strict-transport-security": "max-age=63072000", "transfer-encoding": "chunked", "url": "https://api.vultr.com/v2/instances/d6c5aa04-be2d-4684-b5cf-56bd41c97b9c", "x-content-type-options": "nosniff", "x-frame-options": "DENY", "x-robots-tag": "noindex,noarchive", "x-user": "ansible+opensource@vultr.com"}, "instance": {"hostname": "info2", "label": "ansible-test-55941776-fv-az186-460_info2", "os": "Debian 11 x64 (bullseye)", "plan": "vc2-1c-1gb", "region": "cdg", "tags": ["three", "four"]}, "msg": "Failure while calling the Vultr API v2 with DELETE for \"/instances/d6c5aa04-be2d-4684-b5cf-56bd41c97b9c\"."}
 ```

**Comment by resmo** [comment]
```
fatal: [testhost]: FAILED! => {"changed": false, "msg": "No Resource /applications with deploy_name found: Gitea on Ubuntu 20.04"}
```

**Comment by resmo** [comment]
@happytreees could you set a secret VULTR_API_KEY accessible from this repo for me please.

**Comment by resmo** [comment]
fixed by #84

**Comment by resmo** [comment]
@happytreees a weekly github workflow is going to clean up left over resources created by ansible.
